spectator-api/src/main/java/com/netflix/spectator/api/histogram/PercentileDistributionSummary.java

```diff
@@ -57,19 +73,126 @@
+    return new IdBuilder<Builder>(registry) {
+      @Override protected Builder createTypeBuilder(Id id) {
+        return new Builder(registry, id);
+      }
+    };
+  }
+
+  /**
+   * Helper for getting instances of a PercentileDistributionSummary.
+   */
+  public static final class Builder extends TagsBuilder<Builder> {
+
+    private Registry registry;
+    private Id baseId;
+    private float accuracy;
+    private long max;
+
+    /** Create a new instance. */
+    Builder(Registry registry, Id baseId) {
+      super();
+      this.registry = registry;
+      this.baseId = baseId;
+      this.accuracy = 0.1f;
+      this.max = Long.MAX_VALUE;
+    }
+
+    /**
+     * Sets the desired accuracy for the percentile approximation when an explicit threshold is
+     * set ({@link #withThreshold(long)}). The accuracy flag is used to help reduce the
+     * cost of the percentile approximation by trading off accuracy for percentiles that are
+     * further away from the known SLA or failure threshold.
+     *
+     * @param accuracy
+     *     Value from 0.0 (least accurate) to 1.0 (most accurate). Default is 0.1.
+     * @return
+     *     This builder instance to allow chaining of operations.
+     */
+    public Builder withAccuracy(float accuracy) {
+      if (accuracy < 0.0f || accuracy > 1.0f) {
+        // Invalid value provided, use default.
+        IllegalArgumentException e = new IllegalArgumentException(
+            "Invalid accuracy value for PercentileTimer [" + baseId + "]. Expected value"
+                + " between 0 and 1, received " + accuracy + ".");
+        registry.propagate(e);
+        this.accuracy = 0.1f;
+      } else {
+        // Use the user selection.
+        this.accuracy = accuracy;
+      }
+      return this;
+    }
+
+    /**
+     * Sets the threshold for this summary. The threshold is should be the SLA boundary or
+     * failure point for the activity. Explicitly setting the threshold allows us to optimize
+     * for the important range of values and reduce the overhead associated with tracking the
+     * data distribution.
+     *
+     * For example, suppose you are making a client call and the max payload size is 8mb. Setting
+     * the threshold to 8mb will restrict the possible set of buckets used to those approaching
+     * the boundary. So we can still detect if it is nearing failure, but percentiles
+     * that are further away from the threshold may be inflated compared to the actual value.
+     *
+     * @param amount
+     *     Amount indicating the threshold for this summary.
+     * @return
+     *     This builder instance to allow chaining of operations.
+     */
+    public Builder withThreshold(long amount) {
+      max = amount;
+      return this;
+    }
+
+    /**
+     * Create or get an instance of the percentile distribution summary with the specified
+     * settings.
+     */
+    public PercentileDistributionSummary build() {
```

Comment:
Is it possible to refactor this to remove the duplication with `PercentileTimer#build()`?

Reply:
Probably, but it is an internal detail that we can change without impacting the API. So I'm not too worried about it right now.